microsvcs/user/app/controller/profile_con.js

```diff
@@ -7,33 +8,30 @@ const QUERIES = require("../model/profile.js")
  * 1 request parameter expected.
  * @param {String} uid - unique uid field
  */
-exports.getProfileByUID = async (req, res) => {
+exports.getProfileByUID = async (req, res, next) => {
     try {
         const uid = req.params.uid;
         const result = await QUERIES.findByUID(uid)
-        res.status(200).send(result);
-    } catch (error) {
-        if (error.message == "Server Error Occurred") {
-            res.status(500).send({ message: error.message });
-        } else {
-            res.status(404).send({ message: error.message });
+        if (result === null) {
+            throw new Error("No Reference Found")
         }
+        next(UTIL.formatRes(true, result))
+    } catch (error) {
+        next(UTIL.formatRes(false, error.message))
     };
 };
 
-exports.updateProfileInfo = async (req, res) => {
+exports.updateProfileInfo = async (req, res, next) => {
     try {
         const uid = req.params.uid;
         const json_body = req.body
         const result = await QUERIES.updateOneByUID(uid, json_body)
         if (result.modifiedCount) {
-            res.status(200).send({ message: "success" });
-        }
-    } catch (error) {
-        if (error.message == "Server Error Occurred") {
-            res.status(500).send({ message: error.message });
+            next(UTIL.formatRes(true, result))
         } else {
-            res.status(404).send({ message: error.message });
+            throw new Error("No Reference Found")
         }
+    } catch (error) {
```

Comment:
I think the error handling is supposed to catch automatically without you calling next